jetty-ee10/jetty-ee10-proxy/src/main/java/org/eclipse/jetty/ee10/proxy/AsyncMiddleManServlet.java

```diff
@@ -640,7 +645,10 @@ public void onWritePossible() throws IOException
                         return;
                 }
 
-                this.chunk = chunk = chunks.poll();
+                try (AutoLock ignored = lock.lock())
+                {
+                    this.chunk = chunk = chunks.poll();
+                }
```

Comment:
Shouldn't the lock be released after `writeProxyResponseContent`? I am not sure, but it seems that two threads may poll a chunk(ie there were two chunks in the queue) and then they may race and write the bytes out of order.